pygmt/helpers/utils.py

```diff
@@ -143,6 +144,15 @@ def _validate_data_input(
                     raise GMTInvalidInput(msg)
                 if hasattr(data, "data_vars") and len(data.data_vars) < 3:  # xr.Dataset
                     raise GMTInvalidInput(msg)
+        if kind == "vectors" and isinstance(data, dict):
```

Comment:
Please note that, these new codes in `_validate_data_input` is temporary and I plan to refactor `_validate_data_input` in a separate PR.